examples/PurePursuitController/purePursuitControllerExample.cc

```diff
@@ -0,0 +1,146 @@
+/*
+    Small example program to demonstrate the Pure Pursuit Controller. The controller
+    takes a list of coordinates as waypoint and calculates a desired turning angle
+    to steer the car for a smooth path tracking
+*/
+
+// BoB robotics includes
+#include "common/main.h"
+#include "robots/simulatedAckermanCar.h"
+#include "robots/car_display.h"
+#include "robots/purePursuitController.h"
+
+// Third-party includes
+#include "third_party/units.h"
+
+// Standard C++ includes
+#include <chrono>
+#include <iostream>
+#include <thread>
+
+using namespace BoBRobotics;
+using namespace std::literals;
+using namespace units::literals;
+using namespace units::length;
+using namespace units::angle;
+
+int bob_main(int, char **)
+{
+
+    Robots::SimulatedAckermanCar<> car(1.4_mps, 500_mm);   // simulated car
+    Robots::CarDisplay display(10.2_m,160_mm);             // For displaying the agent
+    std::vector<SDL_Rect> rekt_list;                       // list of waypoints
+
+    float currentX = 0, currentY = 0;                      // current mouse coordinate click
+
+    // adding the first coordinate
+    //-------------------------------------------------------------------
+    millimeter_t xMM, yMM;
+    display.pixelToMM(currentX, currentY,  xMM,  yMM);
+    std::vector<Vector2<millimeter_t>> wpCoordinates; 
+    Vector2<millimeter_t> firstCoord(xMM, yMM);
+    wpCoordinates.push_back(firstCoord);
+    //-------------------------------------------------------------------
+
+    auto mmps = 0_mps;    // speed of robot car
+    degree_t deg = 0_deg; // angle of steering of robot car
+    millimeter_t lookaheadDistance = 1000_mm; // lookahead distance
+
+    auto wheelBase = car.getDistanceBetweenAxis(); // distance between wheel bases
+    BoBRobotics::Robots::PurePursuitController controller(lookaheadDistance, wheelBase);
+
+    while(display.isOpen()) {
+        
+        // each click will spawn a way point which is connected together 
+        std::vector<float> mousePos = display.getMouseClickPixelPosition();
+        float xp = mousePos[0];
+        float yp = mousePos[1];
+        
+        // click to the screen to store waypoint in list
+        if (xp != currentX && yp != currentY) {
+            // draw a rectangle at the goal position
+            SDL_Rect rekt;
+            rekt.x = xp;
+            rekt.y = yp;
+            rekt.w = 5;
+            rekt.h = 5;
+            currentX = xp;
+            currentY = yp;
+            rekt_list.push_back(rekt); // save rectangle
+            
+            // we store the physical unit coordinate in the waypoint list
+            display.pixelToMM(xp, yp,  xMM,  yMM);
+            Vector2<millimeter_t> coords(xMM,yMM);
+            wpCoordinates.push_back(coords);  
```

Comment:
You should be able to do these last two lines in one:
``wpCoordinates.emplace_back(xMM,yMM)``